### PR TITLE
Redoes how stormtrooper aim works, makes it more consistently random

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

~~Making alt prs is a great idea am I right guys?~~
alt of #58287 and #58288
I think you had some thoughts on this @ninjanomnom, lemme know what you think yeah?

The old method was simply varying the spread angle, but that lead to you hitting the target too often, so
blinding was added in a manner that lead to pain at high td.

Then we switched over to using range(2, target), which works about as well as the old spread angle method, and
also had you shooting at yourself/the pipes in the floor.

Both of these methods have the same goal, prevent you from hitting your target outside of hyper close range. So
instead of picking a random close target or changing just the max spread angle, why don't we set a lowest value
you can roll for spread? That way we have some control over the randomness, and we can make the trait work as it
should.

As a part of this, I've changed a few rand()s to rand(0, 1). They were rolling floats between 0 and 1, so they
could in theory roll a 0.5, get 0.5 subtracted from them, and end up producing an angle change of 0. This also
lead to silly behavior with guns not deviating nearly as much as I wanted them to. I'm honestly not sure if this
was intentional or not, but it's dumb when combined with ALREADY ROLLING RANDOM NUMBERS FOR SPREAD
AHHHHHHHHHHHHH

Oh and I haven't readded a blur, since I'm not confident in my artistic ability. If you'd like I can give it a
go, or just use filters or something IDK

## Why It's Good For The Game

Makes stormtrooper aim more consistently bad, cleans up some sin in gun code

## Changelog
:cl:
balance: Stormtrooper aim is now more consistently bad, the same can somewhat be said of anything that affects spread, but it's random already so who knows
/:cl:

### CODE REVIEW

**Comment by ArcaneDefence** [comment]
This would address the discrepancies in https://github.com/tgstation/tgstation/issues/58049

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -303,7 +303,7 @@
 				to_chat(user, "<span class='warning'>[src] is lethally chambered! You don't want to risk harming anyone...</span>")
 				return
 		if(randomspread)
-			sprd = round((rand() - 0.5) * DUALWIELD_PENALTY_EXTRA_MULTIPLIER * (randomized_gun_spread + randomized_bonus_spread))
+			sprd = round((rand(0, 1) - 0.5) * DUALWIELD_PENALTY_EXTRA_MULTIPLIER * (randomized_gun_spread + randomized_bonus_spread))
```
Comment (Mothblocks) [inline code/modules/projectiles/gun.dm:306]: Eh? `rand()` is already `rand(0,1)`.

File: code/modules/projectiles/gun.dm [hunk 2]
```diff
@@ -356,7 +360,7 @@
 				if(chambered.harmful) // Is the bullet chambered harmful?
 					to_chat(user, "<span class='warning'>[src] is lethally chambered! You don't want to risk harming anyone...</span>")
 					return
-			sprd = round((rand() - 0.5) * DUALWIELD_PENALTY_EXTRA_MULTIPLIER * (randomized_gun_spread + randomized_bonus_spread))
+			sprd = round((rand(0, 1) - 0.5) * DUALWIELD_PENALTY_EXTRA_MULTIPLIER * (randomized_gun_spread + randomized_bonus_spread))
```
Comment (Mothblocks) [inline code/modules/projectiles/gun.dm:363]: Same here.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/gun.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/projectiles/gun.dm:306]: There's a key difference between 0,1 and between 0 and 1. One goes by integers, the other by floats
image

File: code/modules/projectiles/gun.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/modules/projectiles/gun.dm:363]: see above

**Review by Mothblocks — APPROVED** [review]

**Pull request merged**